ci-operator/step-registry/configure-registry-storage/deploy-s3-cloudfront/configure-registry-storage-deploy-s3-cloudfront-commands.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# This script will deploy an aws s3 bucket and cloudfront distribution.
+# It need deploy against aws platform.
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    echo "Setting proxy"
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+case "${CLUSTER_TYPE}" in
+aws|aws-arm64|aws-usgov) export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred;;
+aws-c2s|aws-sc2s) export AWS_SHARED_CREDENTIALS_FILE=${SHARED_DIR}/aws_temp_creds;;
+*) echo "Unsupported cluster type '${CLUSTER_TYPE}'"; exit 1;;
+esac
+
+AWS_ACCESS_KEY_ID=$(cat "${AWS_SHARED_CREDENTIALS_FILE}" | grep aws_access_key_id | tr -d ' ' | cut -d '=' -f 2) && export AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY=$(cat "${AWS_SHARED_CREDENTIALS_FILE}" | grep aws_secret_access_key | tr -d ' ' | cut -d '=' -f 2) && export AWS_SECRET_ACCESS_KEY
+
+if [[ ${AWS_ACCESS_KEY_ID} == "" ]] || [[ ${AWS_SECRET_ACCESS_KEY} == "" ]]; then
+  echo "Did not find AWS credential, exit now"
+  exit 1
+fi
+
+# get region to deploy new bucket
+AWS_REGION=${LEASED_RESOURCE} && export AWS_REGION
+NEW_BUCKET=$(head /dev/urandom | tr -dc a-z | head -c 10) && export NEW_BUCKET
+
+# create file to deploy cloudfront using new bucket
+cat >>"${SHARED_DIR}/create_s3_bucket_for_registry_storage.tf" <<EOF
```

Comment:
if `create_s3_bucket_for_registry_storage.tf` is not required in destroy step, I would suggest saving it in `/tmp` and deleting it after this step is complete since it contains the key and credential.

Reply:
@yunjiang29 is need in destroy step, I test locally without create_s3_bucket_for_registry_storage.tf file, destroy failed with error
```
**│ Error: Invalid provider configuration
│
│ Provider "registry.terraform.io/hashicorp/aws" requires explicit configuration. Add a provider block to the root module and configure the
│ provider's required arguments as described in the provider documentation.**
```

Reply:
OK, see https://github.com/openshift/release/pull/38033/files#r1166707790